### PR TITLE
Add option to not use gemini-scrollbar

### PR DESCRIPTION
**Pull request by daiyiwhale**
This PR allows you to opt out of gemini-scrollbar in Dropdown and SidePanel. Gemini-scrollbar will still be used by default.

Incidentally, scrolling on SidePanel will now not cause the background to scroll. (Edit: previously, when the sidepanel was active, scrolling would scroll to the bottom of the sidepanel contents and then begin scrolling the background page contents. I fixed it so that it's not possible to scroll the background page content while the sidepanel is active.)

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/SidePanel/SidePanelContents.js [hunk 1]
```diff
@@ -16,6 +16,14 @@ class SidePanelContents extends Util.mixin(BindMixin) {
     ];
   }
 
+  componentWillReceiveProps(newProps) {
```
Comment (kennyt) [inline src/SidePanel/SidePanelContents.js]: We generally call this `nextProps` :)

Comment (daiyiwhale) [inline src/SidePanel/SidePanelContents.js]: argh that's what I get for copypasta (':

Comment (daiyiwhale) [inline src/SidePanel/SidePanelContents.js]: done!

File: docs/src/Dropdown/index.js [hunk 2]
```diff
@@ -337,6 +337,11 @@ class DropdownExample extends React.Component {
   transition: React.PropTypes.bool,
   // The prefix of the transition classnames.
   transitionName: React.PropTypes.string,
+  // Transition lengths
+  transitionEnterTimeout: React.PropTypes.number,
+  transitionLeaveTimeout: React.PropTypes.number,
+  // Optional disable Gemini scrollbar. Defaults to true.
```
Comment (mlunoe) [inline docs/src/Dropdown/index.js]: Maybe instead: "Option to use Gemini scrollbar. Defaults to true"

**Comment by mlunoe** [comment]
What does this mean?
"Incidentally, scrolling on SidePanel will now not cause the background to scroll."

**Comment by mlunoe** [comment]
Modals text is overflowing instead of scrolling for me:

**Comment by daiyiwhale** [comment]
@mlunoe previously, when the sidepanel was active, scrolling would scroll to the bottom of the sidepanel contents and then begin scrolling the background page contents. I fixed it so that it's not possible to scroll the background page content while the sidepanel is active.

**Comment by daiyiwhale** [comment]
Oh no sad modal ):

**Comment by kennyt** [comment]
resizing the modal when `useGemini` is false also behaves weirdly

**Comment by mlunoe** [comment]
I can't scroll in the sidepanel at all anymore :(

**Comment by mlunoe** [comment]
Ok, side panel scroll problem was not introduced here. I created a separate PR for the fix here: https://github.com/mesosphere/reactjs-components/pull/187
(team effort with @jfurrow @daiyiwhale!)

**Comment by daiyiwhale** [comment]
I'll create a separate PR for the resizing of Modal, as the issues were introduced elsewhere (https://github.com/mesosphere/reactjs-components/pull/174)

**Comment by kennyt** [comment]
text overflow for modal is still a problem atm, correct?

**Comment by daiyiwhale** [comment]
I have a separate pr for the modal fixes

**Comment by daiyiwhale** [comment]
Ah ok, I remember now. The reason we allow it to be possible for the modal to overflow is because we wanted it to be possible for components such as Dropdown to expand past the modal borders.

**Comment by daiyiwhale** [comment]
Alright, when gemini is turned off we're catching the overflow in a native scrollbar. Users will have to manually turn off overflow for now if they want stuff like Dropdown to overflow.

File: docs/src/Dropdown/index.js (hunk 2 above)
Comment (daiyiwhale) [inline docs/src/Dropdown/index.js]: done!

File: docs/styles/page-content.less [hunk 3]
```diff
@@ -32,6 +32,9 @@ Variables
   overflow: hidden;
 }
 
+.visible-overflow {
+  overflow: auto;
+}
```
Comment (jfurrow) [inline docs/styles/page-content.less]: We don't need this class anymore, right?

Comment (daiyiwhale) [inline docs/styles/page-content.less]: oh right! I thought we did it together, but it was lost to chaos :fallen_leaf:

**Pull request merged**